### PR TITLE
feat(itinerary-body): add optional param to itinerary for collapsing alerts

### PR DESCRIPTION
**Pull request by mcleinman**
This adds an optional param to itineraries, which would force alerts to always open in collapsed form, even if there are 1 or 2 alerts. It also turns a magic number in the code into a named variable, and (hopefully) increases the readability of this code by adding some booleans into some not-immediately-intuitive logic. Also added some stories to storybook to cover the various permutations of this new feature.

Once this goes through review, we'd like to use this functionality at TriMet.

### CODE REVIEW

File: packages/itinerary-body/src/TransitLegBody/index.tsx
Comment (miles-grant-ibigroup) [inline packages/itinerary-body/src/TransitLegBody/index.tsx:126]:
```suggestion
      (leg.alerts?.length > maximumAlertCountToShowUncollapsed ||
```

You should be able to write this. It will be compiled down to more browser-compatible, verbose code upon packaging.

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Aside from one code-nitpick this looks really good thanks for cleaning this up! 

I have one issue which is that the stories "Three alerts always collapsing" and "Three alerts not always collapsing" render the exact same output (the alerts are collapsed by default)

Is this expected behavior?

File: packages/itinerary-body/src/TransitLegBody/index.tsx
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/index.tsx:126]: There is a second instance of the same refactoring a few lines below.

**Comment by miles-grant-ibigroup** [comment]
You'll have to run `yarn unit -u` to update the snapshots so that the ci tests pass!

**Comment by mcleinman** [comment]
> I have one issue which is that the stories "Three alerts always collapsing" and "Three alerts not always collapsing" render the exact same output (the alerts are collapsed by default)
> 
> Is this expected behavior?

Yes, it's expected. (All three zero alerts stories also render identical outputs.) I wanted to test 9 combos (3 param types x 3 alert amnts): [true param, false param, no param] x [0 alerts, an amount of alerts that default to expanded (2), an amount of alerts that default to collapsed (3)]. 

While many of these give identical results, I wanted to easily test the logic of each. (If you'd prefer another way to do this, or just fewer stories, let me know and I'll happily update it.) For what it's worth, the "true param with zero alerts" helped me in this latest refactor - there was briefly a logic bomb in there that resulted in an error.

File: packages/itinerary-body/src/TransitLegBody/index.tsx
Comment (mcleinman) [reply packages/itinerary-body/src/TransitLegBody/index.tsx]: Thanks! I pulled this out, and while I was in there I updated the booleans to remove some duplicated logic I created (`leg.alerts.length > maximumAlertCountToShowUncollapsed` was used in two spots).

**Comment by mcleinman** [comment]
Some tests are messed up right now, currently fixing.

**Comment by mcleinman** [comment]
The tests should pass now - they do locally. Per discussion on Slack, I needed to get this working on WSL to do so.

**Review by miles-grant-ibigroup — APPROVED** [review]
Tests are passing and everything looks good to me! Thanks for the PR!

File: packages/itinerary-body/src/TransitLegBody/index.tsx
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/TransitLegBody/index.tsx:128]: Is expanding alerts relevant (or desired) if `leg.alerts` is `null`?

File: packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx:52]: Sort props (case insensitive, 2 instances)

File: packages/itinerary-body/src/ItineraryBody/index.tsx
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/ItineraryBody/index.tsx:33]: How about renaming this prop to simply `alwaysCollapseAlerts`? (re-sort props if you decide to rename)

File: packages/itinerary-body/src/TransitLegBody/index.tsx
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/TransitLegBody/index.tsx:123]: Declare this constant outside the class (e.g. above line 57).

File: packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx:178]: I would think just one story with zero alerts is enough to demonstrate that no alert element is rendered at all. The other zero-alert cases seem redundant.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
I recommend renaming the new prop to a shorter name. Also see other minor comments.

File: packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx
Comment (mcleinman) [reply packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx:178]: While they give identical results, I wanted to easily test the logic of each. (If you'd prefer another way to do this, or think the tradeoff is not worthwhile, let me know and I'll happily update it.) For what it's worth, the "true param with zero alerts" helped me in the last commit - there was briefly a logic bomb in there that resulted in an error that was not present with a false param.

File: packages/itinerary-body/src/TransitLegBody/index.tsx
Comment (mcleinman) [reply packages/itinerary-body/src/TransitLegBody/index.tsx:128]: Yes, in fact it's needed - if `shouldOnlyShowAlertsExpanded` is false, we show the alert toggle. Can rename the variable (or leave a comment) if you'd like that to be a bit clearer.

File: packages/itinerary-body/src/TransitLegBody/index.tsx
Comment (mcleinman) [reply packages/itinerary-body/src/TransitLegBody/index.tsx]: Done, thanks.

File: packages/itinerary-body/src/ItineraryBody/index.tsx
Comment (mcleinman) [reply packages/itinerary-body/src/ItineraryBody/index.tsx]: Good call, updated everywhere. Thanks.

File: packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx
Comment (mcleinman) [reply packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx]: Thanks for that - updated.

File: packages/itinerary-body/src/TransitLegBody/index.tsx
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/TransitLegBody/index.tsx]: A comment about expanding alerts even if `leg.alerts` is `null` would be useful, please.

**Review by binh-dam-ibigroup — APPROVED** [review]
Approving the code, but I will mark this PR blocked until #436 is merged, so that type checks in the CI tests pass.

File: packages/itinerary-body/src/TransitLegBody/index.tsx
Comment (mcleinman) [reply packages/itinerary-body/src/TransitLegBody/index.tsx]: Good call, thanks. I've added a comment. Please let me know if you find it unclear, and want tweaks to it.

File: packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx
Comment (mcleinman) [reply packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx:180]: Done, sorry about that.

File: packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx
Comment (mcleinman) [reply packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx:52]: Done, sorry about that.

**Comment by mcleinman** [comment]
https://github.com/opentripplanner/otp-ui/pull/436 is merged into this branch.

**Pull request merged**
